packages/devtools_app/lib/src/network/network_controller.dart

```diff
@@ -0,0 +1,204 @@
+      }
+    }
+
+    // Build our list of network requests from the collected events.
+    for (final request in httpEvents.entries) {
+      final requestId = request.key;
+      final requestData = HttpRequestData.fromTimeline(
+          _timelineMicrosOffset, request.value);
+
+      // If there's a new event which matches a request that was previously in
+      // flight, update the associated HttpRequestData.
+      if (outstandingRequestsMap.containsKey(requestId)) {
+        final outstandingRequest = outstandingRequestsMap[requestId];
+        outstandingRequest.merge(requestData);
+        if (!outstandingRequest.inProgress) {
+          outstandingRequestsMap.remove(requestId);
+        }
+        continue;
+      } else if (requestData.inProgress) {
+        outstandingRequestsMap.putIfAbsent(requestId, () => requestData);
+      }
+      currentValues.add(requestData);
+    }
+    // Trigger refresh.
+    _httpRequestsNotifier.value = HttpRequests(
+        requests: currentValues, outstanding: outstandingRequestsMap);
+  }
+
+  void _startPolling() {
+    Future<void>.delayed(const Duration(milliseconds: 1000)).then((_) {
+      if (_httpRecordingNotifier.value) {
+        refreshRequests();
+        _startPolling();
+      }
+    });
+  }
+
+  Future<void> _forEachIsolate(Future Function(IsolateRef) callback) async {
+    final vm = await serviceManager.service.getVM();
+    final futures = <Future>[];
+    for (final isolate in vm.isolates) {
+      futures.add(callback(isolate));
+    }
+    await Future.wait(futures);
+  }
+
+  Future<void> _setHttpTimelineRecording(bool state) async {
+    await _forEachIsolate((isolate) async {
+      final future = serviceManager.service
+          .setHttpEnableTimelineLogging(isolate.id, state);
+      // If the isolate is paused the request above will never complete.
+      await Future.any([
+        future,
+        Future.delayed(const Duration(milliseconds: 500)),
+      ]);
+    });
+    _httpRecordingNotifier.value = state;
+  }
+
+  /// Force refreshes the HTTP requests logged to the timeline.
+  Future<void> refreshRequests() async {
+    final timestamp = await serviceManager.service.getVMTimelineMicros();
+    final timeline = await serviceManager.service.getVMTimeline(
+        timeOriginMicros: _lastProfileRefreshMicros,
+        timeExtentMicros: timestamp.timestamp - _lastProfileRefreshMicros);
+    _lastProfileRefreshMicros = timestamp.timestamp;
+    _processHttpTimelineEvents(timeline);
+  }
+
+  /// Enables HTTP request recording on all isolates and starts polling.
+  /// 
+  /// If `alreadyRecording` is true, the last refresh time will be assumed to
+  /// be the beginning of the process (time 0).
+  Future<void> startRecording({
+    bool alreadyRecording = false,
+  }) async {
+    // Set the current timeline time as the time of the last refresh.
+    final timestamp = await serviceManager.service.getVMTimelineMicros();
+
+    if (!alreadyRecording) {
```

Comment:
If we're already recording, should we just return from this method as a no-op?

Reply:
I don't think so. `alreadyRecording` is only `true` when we've determined that HTTP logging is enabled on *at least one isolate*. If that's the case, we can't update the last profile refresh time since we don't know when profiling started, and we need to enable logging on the rest of the isolates (I'm happy to take input on this decision in particular).